### PR TITLE
managarr: init at v0.4.2

### PR DESCRIPTION
**Pull request by IncredibleLaser**
Managarr is a TUI for the *arr stack (currently only supporting Sonarr and Radarr). Hosted at https://github.com/Dark-Alex-17/managarr

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/managarr/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  nativeBuildInputs = [ perl ];
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
+
+  meta = {
+    description = "A TUI and CLI to manage your Servarrs";
+    homepage = "https://github.com/Dark-Alex-17/managarr";
+    license = lib.licenses.mit;
+    maintainers = [ maintainers.IncredibleLaser];
```
Comment (JohnRTitor) [inline pkgs/by-name/ma/managarr/package.nix:33]:
```suggestion
    maintainers = [ lib.maintainers.IncredibleLaser ];
```

Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix]: Thanks for spotting it, fixed

File: pkgs/by-name/ma/managarr/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-sZDkvIVFCilXNRhoeGudoPE+/62yIMElFoOx+XESWOA=";
+  };
+
+  cargoHash = "sha256-owXHTMf7aI/QHUCls0udzpPXTCqB5urMPvrMBtQFCXM=";
+
+  nativeBuildInputs = [ perl ];
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
```
Comment (JohnRTitor) [inline pkgs/by-name/ma/managarr/package.nix:27]: Does it support updating cargoHash as well?

Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix:27]: It's a question I have asked myself as well, but looking at generic-updater.nix, probably no.

I'll look into nix-update and commit again when I've looked into it and can confirm it's working.

Thanks for the time and effort

Comment (JohnRTitor) [reply pkgs/by-name/ma/managarr/package.nix]: Nix-update supports updating cargoHash

Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix]: Thinking about it, I'm not sure there is an easy way to quickly calculate a cargoHash or I'm just not reading correctly. `nix-update` mentions it, but I couldn't make it work locally, it would just update the version number.

Quickly glancing through some Rust packages, I don't see any making use of updateScript, so I have removed it.

Regarding the commit history, is it fine to keep it like this or should commits be squashed? I don't know if there's a consensus in nixpkgs.

Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix]: > Nix-update supports updating cargoHash

OK, thanks. For the current version, I'll leave it as is and include nix-update when the next package update hits.

File: pkgs/by-name/ma/managarr/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+let
+  pname = "managarr";
+  version = "0.4.2";
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version;
+  src = fetchFromGitHub {
+    owner = "Dark-Alex-17";
+    repo = pname;
```
[the remark this answers is not in the document]
Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix:17]: I don't think this is correct, if anything it'd be `repo = "managarr";` as a string, however currently I'm using the variable. Should it be changed to a fixed string?

[the remark this answers is not in the document]
Comment (JohnRTitor) [reply pkgs/by-name/ma/managarr/package.nix:17]: It should be string, I apologize for the typo. My point is we should not reuse `pname` anywhere else (unless it's a composite package requiring pname in its steps), as it would break the build if pname is overriden.

File: pkgs/by-name/ma/managarr/package.nix (hunk 2 above)
Comment (JohnRTitor) [reply pkgs/by-name/ma/managarr/package.nix]: > Regarding the commit history, is it fine to keep it like this or should commits be squashed?

The commits should be squashed, but if someone is unable to do so we can squash merge from GitHub to squash into a single commit. 

> For the current version, I'll leave it as is and include nix-update when the next package update hits

You do you 🤷🏻‍♂️

File: pkgs/by-name/ma/managarr/package.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix]: Thanks for the explanation. I have changed it to the fixed string.

I have also included the update script and squashed all commits into one. Not sure setting the script explicitly is necessary, the information is a bit sparse.

File: pkgs/by-name/ma/managarr/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "managarr";
+  version = "0.4.2";
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version;
+  src = fetchFromGitHub {
+    owner = "Dark-Alex-17";
+    repo = "managarr";
+    rev = "v${version}";
```
Comment (IncredibleLaser) [inline pkgs/by-name/ma/managarr/package.nix:19]: I'm a bit unsure still about using ref the way I do with the tag directly. It works, as this is how I built the package, but I just see that usually it's something like `/revs/tags/${version}` (nix-update doesn't seem to be able to use the tag attribute yet). If necessary, I can change this.

Comment (lucasew) [reply pkgs/by-name/ma/managarr/package.nix:19]: Use gitUpdater instead of nix-update-script, then you only need to specify rev-prefix

Ex: `passthru.updateScript = gitUpdater { rev-prefix = "v"; };`

File: pkgs/by-name/ma/managarr/package.nix (hunk 2 above)
Comment (lucasew) [reply pkgs/by-name/ma/managarr/package.nix]: > Thinking about it, I'm not sure there is an easy way to quickly calculate a cargoHash or I'm just not reading correctly. `nix-update` mentions it, but I couldn't make it work locally, it would just update the version number.
> 
> Quickly glancing through some Rust packages, I don't see any making use of updateScript, so I have removed it.
> 
> Regarding the commit history, is it fine to keep it like this or should commits be squashed? I don't know if there's a consensus in nixpkgs.

In my experience, at least in node packages, that use an approach kinda similar, `gitUpdater` updates all these hashes just fine. I'd recommend you start packaging the version just before the last one then when you have something working then test with `maintainers/scripts/update.nix`.

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

- **Extra args**: ``
- **Free space**: true
- **Who asked**: lucasew

**TARGETS**
- **x86_64-linux**: true
- **x86_64-darwin**: false
- **aarch64-darwin**: false

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12448058129

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367042`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>managarr</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -9699,6 +9699,12 @@
     githubId = 36667224;
     name = "Yingchi Long";
   };
+  IncredibleLaser = {
+    github = "IncredibleLaser";
+    githubID = 45564436;
```
Comment (IncredibleLaser) [inline maintainers/maintainer-list.nix:9704]: needs to be githubId

File: pkgs/by-name/ma/managarr/package.nix (hunk 4 above)
Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix:19]: Thanks for the input; for the moment, I've left it at nix-update-script in order not to change too much, and the only example of a Rust package using autoupdate I could find quickly was kanidm, which uses it to specify some options. I assume the bot runs update scripts on all packages automatically anyways so stating this here isn't strictly necessary.

Also it's a bit opaque to me how the hashes are updated, I can run the update scripts, but it will only update the versions here, but hashes remain unchanged, so there needs to be something happening in the background that isn't explained where I have looked at yet.

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -9699,6 +9699,12 @@
     githubId = 36667224;
     name = "Yingchi Long";
   };
+  IncredibleLaser = {
+    github = "IncredibleLaser";
+    githubId = 45564436;
+    matrix = "@incrediblelaser:tchncs.de";
+    name = "Gereon Schomber";
+  };
```
Comment (ambroisie) [inline maintainers/maintainer-list.nix:9707]: Should be split in a different commit: `maintainers: add IncredibleLaser`.

File: pkgs/by-name/ma/managarr/package.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  rustPlatform,
+  nix-update-script,
+  perl,
+}:
+
+let
+  pname = "managarr";
+  version = "0.4.2";
+in
+
```
Comment (ambroisie) [inline pkgs/by-name/ma/managarr/package.nix:13]: Useless `let` block, please inline those.

File: pkgs/by-name/ma/managarr/package.nix (hunk 4 above)
Comment (ambroisie) [inline pkgs/by-name/ma/managarr/package.nix:19]:
```suggestion
    tag = "v${version}";
```

File: pkgs/by-name/ma/managarr/package.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  nativeBuildInputs = [ perl ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "A TUI and CLI to manage your Servarrs";
+    homepage = "https://github.com/Dark-Alex-17/managarr";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.IncredibleLaser ];
```
Comment (ambroisie) [inline pkgs/by-name/ma/managarr/package.nix:33]: Add `mainProgram = "managarr"`.

**Review by ambroisie — DISMISSED** [review]

File: pkgs/by-name/ma/managarr/package.nix (hunk 4 above)
Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix:19]: I left this because of this https://github.com/Mic92/nix-update/issues/304 which this package uses, but can change it.

File: pkgs/by-name/ma/managarr/package.nix (hunk 7 above)
Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix:13]: I can do that, it's not useless though as the version is used twice, and the attribute set isn't recursive (as I thought recursive sets were to be avoided as a general rule of thumb).

File: pkgs/by-name/ma/managarr/package.nix (hunk 8 above)
Comment (IncredibleLaser) [reply pkgs/by-name/ma/managarr/package.nix:26]: Good proposal, done

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (IncredibleLaser) [reply maintainers/maintainer-list.nix:9707]: Done

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12467043157


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367042`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>managarr</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367042`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>managarr</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367042`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>managarr</li>
  </ul>
</details>

**Comment by lucasew** [comment]
Diff LGTM. I didn't test in my machine.

File: pkgs/by-name/ma/managarr/package.nix [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    tag = "v${version}";
+    hash = "sha256-OxGFubtMsGnR8cWDKkeAgryY095uydA3LzE5SS0dspQ=";
+  };
+
+  cargoHash = "sha256-bws4LwpJGfq0hCA2Fq51uCMvr0arbayppq1jvDrILZI=";
+
+  nativeBuildInputs = [ perl ];
+
+  meta = {
+    description = "A TUI and CLI to manage your Servarrs";
```
Comment (pluiedev) [inline pkgs/by-name/ma/managarr/package.nix:23]: Avoid leading articles (see pkgs/README.md)
```suggestion
    description = "TUI and CLI to manage your Servarrs";
```

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (JohnRTitor) [reply maintainers/maintainer-list.nix:9707]: maintainers commit should be the first, I went ahead and did it for you

**Review by JohnRTitor — APPROVED** [review]
